src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,789 @@
+                failedRowsErrors.add(error);
+            }
+        }
+        
+        return failedRowsErrors;
+    }
+
+    /**
+     * Processes a single ACH batch detail, and routes a Payee ACH Account maintenance document accordingly.
+     */
+    protected String processACHBatchDetail(PayeeACHAccountExtractDetail achDetail) {
+        LOG.info("processACHBatchDetail: Starting processACHBatchDetail for: " +achDetail.getLogData());
+        String processingError = null;
+        Person payee = personService.getPersonByPrincipalName(achDetail.getNetID());
+
+        processingError = validateACHBatchDetail(achDetail, payee);
+        if (ObjectUtils.isNotNull(processingError) && StringUtils.isNotBlank(processingError)) {
+            return processingError;
+        }
+        
+        // Check for existing ACH accounts.
+        PayeeACHAccount entityAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.ENTITY, payee.getEntityId(), getDirectDepositTransactionType());
+        PayeeACHAccount employeeAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.EMPLOYEE, payee.getEmployeeId(), getDirectDepositTransactionType());
+        
+        // Add or update Entity ID account.
+        if (ObjectUtils.isNull(entityAccount)) {
+            processingError = addACHAccount(payee, achDetail, PayeeIdTypeCodes.ENTITY);
+            if (ObjectUtils.isNull(processingError)) {
+                LOG.info("processACHBatchDetail: Created new ACH Account of Entity type for payee " + payee.getPrincipalName());
+            } else {
+                return processingError;
+            }
+        } else {
+            processingError = updateACHAccountIfNecessary(payee, achDetail, entityAccount);
+            if (ObjectUtils.isNull(processingError)) {
+                LOG.info("processACHBatchDetail: Any necessary update processing performed to ACH Account of Entity type for payee " + payee.getPrincipalName());
+            } else {
+                return processingError;
+            }
+        }
+        
+        // Add or update Employee ID account.
+        if (ObjectUtils.isNull(employeeAccount)) {
+            processingError = addACHAccount(payee, achDetail, PayeeIdTypeCodes.EMPLOYEE);
+            if (ObjectUtils.isNull(processingError)) {
+                LOG.info("processACHBatchDetail: Created new ACH Account of Employee type for payee " + payee.getPrincipalName());
+            } else {
+                return processingError;
+            }
+        } else {
+            processingError = updateACHAccountIfNecessary(payee, achDetail, employeeAccount);
+            if (ObjectUtils.isNull(processingError)) {
+                LOG.info("processACHBatchDetail: Any necessary update processing performed to ACH Account of Employee type for payee " + payee.getPrincipalName());
+            } else {
+                return processingError;
+            }
+        }
+ 
+        return null;

```

Comment:
Could this return an empty string (there is a KFSConstants.EMPTY_STRING) vs. null?